clients/drcachesim/tools/syscall_mix.h

```diff
@@ -70,6 +70,8 @@ class syscall_mix_t : public analysis_tool_t {
 
 protected:
     struct shard_data_t {
+        // Provide a virtual destructor to allow subclassing.
+        virtual ~shard_data_t() = default;
```

Comment:
nit: perhaps we could add a sentence to the PR description giving a hint on why we have a need for subclassing of `shard_data_t` now.

Reply:
Added a mention of adding callstacks to each syscall, which is my planned usage (there is no memtrace callstack library upstream right now so this would be a separate extension via the subclass at least in the short term).